frontend/src/app/features/in-app-notifications/bell/state/ian-bell.service.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  Injectable,
+  Injector,
+} from '@angular/core';
+import { IanBellStore } from './ian-bell.store';
+import { InAppNotificationsService } from 'core-app/core/state/in-app-notifications/in-app-notifications.service';
+import { IAN_FACET_FILTERS } from 'core-app/features/in-app-notifications/center/state/ian-center.store';
+import {
+  map,
+  tap,
+} from 'rxjs/operators';
+import { Observable } from 'rxjs';
+import { IanBellQuery } from 'core-app/features/in-app-notifications/bell/state/ian-bell.query';
+import {
+  EffectCallback,
+  EffectHandler,
+} from 'core-app/core/state/effects/effect-handler.decorator';
+import { notificationsMarkedRead } from 'core-app/core/state/in-app-notifications/in-app-notifications.actions';
+import { UntilDestroyedMixin } from 'core-app/shared/helpers/angular/until-destroyed.mixin';
+
+@Injectable()
+@EffectHandler
+export class IanBellService extends UntilDestroyedMixin {
```

Comment:
Do we still need to extend from the mixin here?

Reply:
I rewrote the decorator to do the ngOnDestroy hook on our own to properly unsubscribe (There's no way to find out if the service was destroyed other then checking the method call). The Mixin helps for components to unsubscribe more easily, but with my solution the service doesn't need to extend anything.